### ISSUE TITLE
Investigate why test-related SSD's aren't cleaning up

### ISSUE THREAD

**Issue by caugustus-sourcegraph** [sourcegraph/sourcegraph#31262]
https://github.com/sourcegraph/deploy-sourcegraph/tree/master/tests/integration creates live GKE clusters and tears them down after test completion. We noticed that there were a large number of unused test SSD's still running in the CI project.

We should confirm that disks are being correctly deleted after a test run, even if the test fails. Consider adding monitoring for unused disks.

**Comment by caugustus-sourcegraph** [comment]
Disks were retained when I deleted the cluster without deleting the deployment in Kubernetes first. There may be other times that this happens (particularly gitserver/indexed-server/grafana have PVC protection and might not be deleted automatically).

**Comment by danieldides** [comment]
I wonder if this is because we set the `reclaimPolicy: Retain` on the [`StorageClass`](https://docs.sourcegraph.com/admin/install/kubernetes/configure#google-cloud-platform-gcp).

> Should you want to prevent deletion of dynamically provisioned persistent disks, set the reclaim policy of the PersistentVolume resource, or its StorageClass resource, to Retain. In this case, you are charged for the persistent disk for as long as it exists even if there is no PersistentVolumeClaim consuming it. For examples on how to change the reclaim policy, see [Change the Reclaim Policy of a PersistentVolume](https://kubernetes.io/docs/tasks/administer-cluster/change-pv-reclaim-policy/). and [StorageClass Resource](https://kubernetes.io/docs/concepts/storage/storage-classes/#the-storageclass-resource).

https://cloud.google.com/kubernetes-engine/docs/concepts/persistent-volumes

**Comment by caugustus-sourcegraph** [comment]
No abandoned disks seen in the CI project since the tests were updated to delete namespaces. Pricing is probably better handled as a separate task (and maybe already be covered by the third-party provider).

**Comment by daxmc99** [comment]
We need to add alerting for this as it seems to have occurred again

**Comment by michaellzc** [comment]
I think a good way to mitigate this (besides adding alerts) is to enforce every test deployment to create storageclass with retention set to `Delete`.

I found this quite helpful when testing stuff on temp clusters.

**Comment by caugustus-sourcegraph** [comment]
@daxmc99  Do you have more details on what happened? I'd like to create a new issue to track the proposed work instead of reopening this one.

**Comment by daxmc99** [comment]
@caugustus-sourcegraph Not much to describe other than there were a large number of disks looking like `gke-dogfood-952a5131-d-pvc-*` 
These disks are typically created by the K8s PV storage driver. 

I was able to delete all of them with the slightly dangerous `gcloud compute disks list | awk '{print $1}' | xargs gcloud compute disks delete` (which doesn't delete attached disks... at the time of writing).

This isn't an issue for just this project though, I found unattached disks in several places in our projects. I believe the ones I found in this project were related to this issue though.
